src/lib/Libtpp/tpp_transport.c

```diff
@@ -1249,8 +1249,13 @@ add_transport_conn(phy_conn_t *conn)
 				tpp_log(LOG_CRIT, __func__, "Multiplexing failed");
 				return -1;
 			}
-			if (the_post_connect_handler)
-				the_post_connect_handler(fd, NULL, conn->ctx, conn->extra);
+			if (the_post_connect_handler) {
+				if (the_post_connect_handler(fd, NULL, conn->ctx, conn->extra)) {
+					/* e.g.: the tpp_handle_auth_handshake could have failed */
+					tpp_log(LOG_CRIT, __func__, "the_post_connect_handler failed");
+					return -1;
```

Comment:
Is handle_disconnect not required here ?

Reply:
This is in function `add_transport_conn()`. It would cause a double `handle_disconnect()`. The `return -1` is enough here. If `add_transport_conn()` returns != 0, the handle_disconnect() is called here:
![image](https://github.com/openpbs/openpbs/assets/16756616/b5d20291-87b2-4f7d-9313-d3c7cb7f1013)